### PR TITLE
Simplifying the documentation

### PR DESCRIPTION
**Pull request by sushilkm**
Fixes https://github.com/Azure/Azure-Orbital-Analytics-Samples/issues/83

### CODE REVIEW

File: deploy/README.md [hunk 1]
```diff
@@ -1,27 +1,121 @@
...
 
-Follow the [document](./deploy-infrastructure.md) to understand and execute the infrastructure deployment process.
+The deployment script uses following tools, please follow the links provided to install the suggested tools on your computer using which you would execute the script.
 
-## Executing the synapse pipeline
+- [git](https://github.com/git-guides/install-git) 
+- [az cli](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [jq](https://stedolan.github.io/jq/download/)
+- The scripts are executed on bash shell, so if using a computer with windows based operating system, install a [WSL](https://docs.microsoft.com/windows/wsl/about) environment to execute the script.
+- Install bicep on az cli using command `az bicep install`.
```
Comment (karthick-rn) [inline deploy/README.md:20]:
```suggestion
- Install bicep on az cli using command  `az bicep install` and ensure the bicep version is >=0.8.9.
```
and remove the next bullet point

File: deploy/README.md [hunk 2]
```diff
@@ -1,27 +1,121 @@
...
+- The bicep templates have been written to adhere to the syntax and rules for bicep version >= 0.8.9. Please check your bicep version using `az bicep version` if you run into bicep related errors.
+- The user performing the deployment of the bicep template and the associated scripts should have `Owner` role assigned at the subscription to which the resources are being deployed.
```
Comment (karthick-rn) [inline deploy/README.md:22]: Move this line to a separate section after prereqs similar to [this](https://github.com/Azure/Azure-Orbital-STAC/tree/main/deploy#rbac-requirement)

File: deploy/README.md [hunk 3]
```diff
@@ -1,27 +1,121 @@
...
+- This solution assumes no interference from Policies deployed to your tenant preventing resources from being deployed.
```
Comment (karthick-rn) [inline deploy/README.md:23]: we decided to move this line to troubleshooting.md and update as `TODO` so later we can add more details

File: deploy/README.md [hunk 4]
```diff
@@ -1,27 +1,121 @@
...
+```bash
+git clone git@github.com:Azure/Azure-Orbital-Analytics-Samples.git
+```
 
-Follow the [document](./using-pipeline.md#executing-the-pipeline) to configure and execute the pipeline.
+Alternatively, one can use Azure Cloud Bash to deploy this sample solution in their Azure subscription.
+
+## Infrastructure Deployment
+
+Before executing the script one would need to login to azure using `az` cli and set the subscription in which they want to provision the resources.
```
Comment (karthick-rn) [inline deploy/README.md:34]:
```suggestion
Before executing the script, please login to azure using `az` cli and set the subscription in which you want to provision the resources.
```

File: deploy/README.md [hunk 5]
```diff
@@ -1,27 +1,121 @@
...
+
+```bash
+az login
+az account set -s <subscription_id>
+```
+
+To install, configure and package the infrastructure for dependencies one can execute the script [setup.sh](./setup.sh)
```
Comment (karthick-rn) [inline deploy/README.md:41]: Can we update lines from 41 to 54 to something like below:

The following command will install, configure and generate the custom vision model package.

```bash
./deploy/setup.sh <environmentCode> <location>
```
- `environmentCode` used as a prefix for Azure resource names. It allows only alpha numeric(no special characters) and must be between 3 and 8 characters.
- `location` is a valid Azure region.

For eg.
```bash
./deploy/setup.sh aoi eastus
```

File: deploy/README.md [hunk 6]
```diff
@@ -1,27 +1,121 @@
...
+
+For eg.
+```bash
+./deploy/setup.sh aoi eastus
+```
+
+[setup.sh](./setup.sh) executes tasks in 3 steps
+- installs the infrastructure using [install.sh](./install.sh) script.
+- configures the infrastructure for setting up the dependecies using [configure.sh](./configure.sh) script.
+- packages the pipeline code to imported into synapse for standalone executions using [package.sh](./package.sh) script.
```
Comment (karthick-rn) [inline deploy/README.md:59]: I'd just say the package script will generate the custom vision model package and not have reference to importing into Synapse.

File: deploy/README.md [hunk 7]
```diff
@@ -1,27 +1,121 @@
...
+
+**Note**: After the script is completed without errors, please check the batch-account pool created is created succesfully.
```
Comment (karthick-rn) [inline deploy/README.md:61]:
```suggestion
**Note**: After the setup script has run successfully, please check `<envCode>-orc-rg`, select the batch account service and ensure the batch pool has been created successfully.
```

File: deploy/README.md [hunk 8]
```diff
@@ -1,27 +1,121 @@
...
+
+One can find the batch account named as `batchact<10-character-random-string>` in the resource-group `<environment-code>-orc-rg`.
```
Comment (karthick-rn) [inline deploy/README.md:63]:
```suggestion 
You can find the batch account named as `batchact<10-character-random-string>` under the resource group `<environment-code>-orc-rg`.
```

File: deploy/README.md [hunk 9]
```diff
@@ -1,27 +1,121 @@
...
+```bash
+NO_DELETE_DATA_RESOURCE_GROUP=true
+NO_DELETE_MONITORING_RESOURCE_GROUP=true
+NO_DELETE_NETWORKING_RESOURCE_GROUP=true
+NO_DELETE_ORCHESTRATION_RESOURCE_GROUP=true
+NO_DELETE_PIPELINE_RESOURCE_GROUP=true
+./deploy/cleanup.sh <environmentCode>
+```
 
 ## AI Model
```
Comment (karthick-rn) [inline deploy/README.md:113]: Do we still need this section? @mandarinamdar @senthilkungumaraj

**Comment by karthick-rn** [comment]
> Please review only the last commit titled "Updated documentation", the other two commits are already covered in #86

@sushilkm, As we merged #86 we can remove this line I think.

File: deploy/README.md [hunk 10]
```diff
@@ -1,27 +1,121 @@
...
 
-The deployment involves the following steps outlined below:
+We are using bicep templates to deploy the infrastructure resources.
+We will be using following azure services:
+- Azure Batch Account
+- Azure Synapse Analytics
+- Azure Keyvault
+- Azure Storage Account
+- Azure Networking
+- Azure Log Analytics
```
[the remark this answers is not in the document]
Comment (sushilkm) [reply deploy/README.md:10]: I thought of listing here so that user is aware about what bicep template is deploying.

File: deploy/README.md (hunk 9 above)
Comment (mandarinamdar) [reply deploy/README.md:113]: Its already covered in main README, we can remove this from this deploy readme section

Comment (sushilkm) [reply deploy/README.md:113]: @mandarinamdar we decided to remove from the main README

File: docs/Troubleshooting.md [hunk 11]
```diff
@@ -44,3 +44,8 @@ Solution:
 
 After importing your pipeline from your repository or pipeline template approach, please use "Publish" or "Publish All" to publish your pipeline and its associated components like Linked Services.
 
+## Policies interference
+
+This solution assumes no interference from Policies deployed to your tenant preventing resources from being deployed.
+
+TODO: Add more information for this section
```
Comment (senthilkungumaraj) [inline docs/Troubleshooting.md:51]: We need to remove this before merging to main branch

File: deploy/bring-your-own-ai-model.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
+## Using AI-Models
```
Comment (senthilkungumaraj) [inline deploy/bring-your-own-ai-model.md:1]: Can we keep the deploy folder less simplified and with less files? I feel there are too many files under this folder. We should instead move this markdown file to docs. 

@mandarinamdar  Thoughts?

File: azure-pipelines/PR-pipeline.yml [hunk 13]
```diff
@@ -69,7 +69,9 @@ jobs:
         export POSTGRES_ADMIN_LOGIN_PASS=$({ </dev/urandom LC_ALL=C grep -ao '[A-Za-z0-9]' | head -n$((RANDOM % 8 + 9)); echo ${chars:$((RANDOM % ${#chars})):1}; } | shuf | tr -d '\n')
 
         # Setup the infrastructure for testing the pipeline
-        DEPLOY_PGSQL=${USE_POSTGRE_SQL} $(System.DefaultWorkingDirectory)/deploy/setup.sh ${ENV_CODE} eastus custom-vision-model ${ENV_CODE}-environment ${CI_BATCH_ACCOUNT_NAME}
+        DEPLOY_PGSQL=${USE_POSTGRE_SQL} \
+          PRE_PROVISIONED_AI_MODEL_INFRA_NAME=${CI_BATCH_ACCOUNT_NAME} \
+          $(System.DefaultWorkingDirectory)/deploy/setup.sh ${ENV_CODE} eastus ${ENV_CODE}-ci-environment
```
Comment (senthilkungumaraj) [inline azure-pipelines/PR-pipeline.yml:74]: Need to parameterize the location and remove hardcoded value. I understand this is for the pipeline. But if we need to point to a different region for some reason, we need to make sure it does not require a code change.

File: README.md [hunk 14]
```diff
@@ -7,16 +7,9 @@ Disclaimer: The solution and samples provided in this repository is for learning
...
 
-Import the pipeline under the `workflow` folder to your Azure Synapse Analytics instance's workspace. Alternatively, you can copy the files to your repository (git or Azure DevOps) and link the repository to your Azure Synapse Analytics workspace. 
-
-Sample pipelines are provided that include the following AI Model:
-
-### a. AI model
-
-This solution uses the [Custom Vision Model](/src/aimodels) as a sample AI model for demonstrating end to end Azure Synapse workflow geospatial analysis. This sample solution uses Custom Vision model to detect pools in a given geospatial data. 
-You can use any other AI model for object detection or otherwise to run against this solution with a similar [specification](/src/aimodels/custom_vision_object_detection_offline/specs/custom_vision_object_detection.json) or different specification as defined by AI model to integrate in your solution.  
+This solution uses the [Custom Vision Model](/src/aimodels) as a sample AI model to demonstrate end-to-end Azure Synapse workflow for geospatial analysis. In this sample solution, the AI model detects swimming pools for a given geospatial data. You can use this solution to integrate other AI models.
```
Comment (senthilkungumaraj) [inline README.md:12]: Need a line at the end to communicate that using another model in place of the current custom vision model is not a simple replace process. It requires its own geospatial transforms. I would add something like...

"Each AI Model requires their input geospatial data to be in a specific format. When adapting this solution for a different AI Model, make sure the geospatial data transform steps are modified to adapt to the individual AI Model's needs."

File: deploy/bring-your-own-ai-model.md [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+
+There are three ways to load an AI Model with this pipeline. One can use one of the three methods listed below.
+
+### Use the publicly hosted Custom Vision Model as GitHub Packages.
```
Comment (senthilkungumaraj) [inline deploy/bring-your-own-ai-model.md:5]: I think this needs to be generic and not mention Custom Vision Model. The three methods that we documented in the earlier version were to illustrate the extensibility for hosting Custom Vision model.

Also, if our goal is to simplify the documentation, we should only point to the approach where they upload their AI Model container image to the ACR deployed by this solution. 

@mandarinamdar  Thoughts?

**Review by senthilkungumaraj — APPROVED** [review]

File: README.md (hunk 14 above)
Comment (sushilkm) [reply README.md]: done

File: azure-pipelines/PR-pipeline.yml (hunk 13 above)
Comment (sushilkm) [reply azure-pipelines/PR-pipeline.yml]: done.

File: deploy/bring-your-own-ai-model.md (hunk 12 above)
Comment (sushilkm) [reply deploy/bring-your-own-ai-model.md:1]: I would agree on this. May be another change should address this.
We can move the script files to scripts directory and create a docs directory in here to keep context related documents with README pointing to those files.

File: docs/Troubleshooting.md (hunk 11 above)
Comment (sushilkm) [reply docs/Troubleshooting.md:51]: This **TODO** was added in response of an earlier comment.

File: deploy/bring-your-own-ai-model.md (hunk 12 above)
Comment (mandarinamdar) [reply deploy/bring-your-own-ai-model.md:1]: I am fine it getting addressed in a separate PR

File: deploy/bring-your-own-ai-model.md (hunk 15 above)
Comment (mandarinamdar) [reply deploy/bring-your-own-ai-model.md:5]: I think the file name is incorrect. We should be calling the file as using-ai-model-in-pipeline. We are not elaborating here about BYO AI model, that is covered in this doc. But mainly that how to use AI models in a pipeline

File: deploy/README.md (hunk 9 above)
Comment (mandarinamdar) [reply deploy/README.md:113]: In the main README, it is covered under GETTING started. I think that is repeating here and we are adding just last line for bring your own model. But bring your own model is again not covering that alone but covering additional content to use existing model in ACR/github.

File: docs/Troubleshooting.md (hunk 11 above)
Comment (mandarinamdar) [reply docs/Troubleshooting.md:51]: If we need a different PR, lets create an issue to track that vs adding todo here

File: deploy/README.md (hunk 9 above)
Comment (mandarinamdar) [reply deploy/README.md:113]: Lets call out only this part of that section

**Follow the [document](https://github.com/sushilkm/Azure-Orbital-Analytics-Samples/blob/suskuma/doc-fixes/deploy/bring-your-own-ai-model.md) to understand and use custom vision model (Hyperlink) or bring different ai-model for processing with the pipeline.** 

Also lets rename  bring-your-own-ai-model.md to **using-ai-model-in-pipeline.md or something similar**

**Review by mandarinamdar — CHANGES_REQUESTED** [review]
I have left the comments in ongoing conversations

File: deploy/bring-your-own-ai-model.md (hunk 12 above)
Comment (sushilkm) [reply deploy/bring-your-own-ai-model.md]: created the issue https://github.com/Azure/Azure-Orbital-Analytics-Samples/issues/91

File: docs/Troubleshooting.md (hunk 11 above)
Comment (sushilkm) [reply docs/Troubleshooting.md]: created the issue https://github.com/Azure/Azure-Orbital-Analytics-Samples/issues/91

File: deploy/bring-your-own-ai-model.md (hunk 15 above)
Comment (karthick-rn) [reply deploy/bring-your-own-ai-model.md]: I think based on our discussion last week we decided to create issue #88 to cover the instructions on bringing new AI models.

**Review by karthick-rn — APPROVED** [review]

File: deploy/bring-your-own-ai-model.md (hunk 12 above)
Comment (mandarinamdar) [reply deploy/bring-your-own-ai-model.md]: Lets create a issue for this and handle it in separate PR

**Review by mandarinamdar — COMMENTED** [review]
Lets create issue for structuring the Deploy folder better in a separate PR. rest looks good. thanks for addressing the comments

**Review by mandarinamdar — APPROVED** [review]

File: deploy/bring-your-own-ai-model.md (hunk 12 above)
Comment (sushilkm) [reply deploy/bring-your-own-ai-model.md]: Issue created

**Pull request merged**